scripts/deployment-helpers.sh

```
@@ -159,3 +206,105 @@ wait_for_csv() {
+  echo "⚠️  Timed out waiting for ${resource_type}/${resource_name}" >&2
+  return 1
+}
+
+# version_compare <version1> <version2>
+#   Compares two version strings in semantic version format (e.g., "4.19.9")
+#   Returns 0 if version1 >= version2, 1 otherwise
+version_compare() {
+  local version1="$1"
+  local version2="$2"
+  
+  local v1=$(echo "$version1" | awk -F. '{printf "%d%03d%03d", $1, $2, $3}')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)